### PR TITLE
Update RunInference intro

### PR DESCRIPTION
**Pull request by rszper**
- Update the RunInference intro to simplify the sentences
- Add a support and limitations section; move some concepts out of the intro sentence into this section

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by rszper** [comment]
R: @rezarokni

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md [hunk 1]
```diff
@@ -47,9 +47,16 @@ For more infomation about machine learning with Apache Beam, see:
 * [About Beam ML](/documentation/ml/about-ml)
 * [RunInference notebooks](https://github.com/apache/beam/tree/master/examples/notebooks/beam-ml)
 
+## Support and limitations
+
+- The RunInference API is supported in Apache Beam 2.40.0 and later versions.
+- PyTorch and Scikit-learn frameworks are supported. Tensorflow models are supported through tfx-bsl.
```
Comment (rezarokni) [inline website/www/site/content/en/documentation/sdks/python-machine-learning.md:53]: This is out dated. we no longer need tfx-bsl. Lets just change this to Tensorflow, Pytorch and other frame works are supported, for a full list look ...

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md [hunk 2]
```diff
@@ -47,9 +47,16 @@ For more infomation about machine learning with Apache Beam, see:
...
+- The RunInference API supports batch and streaming pipelines.
+- The RunInference API supports local and remote inference.
```
Comment (rezarokni) [inline website/www/site/content/en/documentation/sdks/python-machine-learning.md:55]: local to the runner worker

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md [hunk 3]
```diff
@@ -47,9 +47,16 @@ For more infomation about machine learning with Apache Beam, see:
...
+
 ## Why use the RunInference API?
 
-RunInference takes advantage of existing Apache Beam concepts, such as the `BatchElements` transform and the `Shared` class, to enable you to use models in your pipelines to create transforms optimized for machine learning inferences. The ability to create arbitrarily complex workflow graphs also allows you to build multi-model pipelines.
+RunInference takes advantage of existing Apache Beam concepts, such as the `BatchElements` transform and the `Shared` class, to enable you to use models in your pipelines optimized for machine learning inferences. The ability to create arbitrarily complex workflow graphs also allows you to build multi-model pipelines.
```
Comment (rezarokni) [inline website/www/site/content/en/documentation/sdks/python-machine-learning.md:59]: @damccorm Can we add the full list here as we have done much more since the early days :-) Model manager would be nice to discuss.

**Review by rezarokni — APPROVED** [review]

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md (hunk 1 above)
Comment (rszper) [reply website/www/site/content/en/documentation/sdks/python-machine-learning.md]: Updated, but the update should be verified.

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md (hunk 2 above)
Comment (rszper) [reply website/www/site/content/en/documentation/sdks/python-machine-learning.md]: I updated this, but I don't know what local to the runner worker means, so someone should make sure that what I wrote is correct, and maybe we should clarify.

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md (hunk 3 above)
Comment (rszper) [reply website/www/site/content/en/documentation/sdks/python-machine-learning.md:59]: We're currently updating content in three places (here, https://beam.apache.org/documentation/ml/about-ml/, and the Dataflow ML pages). My preference would be to have this page redirect to https://beam.apache.org/documentation/ml/about-ml/ and move any content to that page.

Comment (damccorm) [reply website/www/site/content/en/documentation/sdks/python-machine-learning.md:59]: +1 on redirecting to that page. We'd probably need to move most of the content from this page over there (and deduplicate it). We need to make sure the link is still valid, but we can do that with [aliases](https://github.com/apache/beam/blob/3d3a7afe0df985f3b1ff7632edbf435bcf86c80c/website/www/site/content/en/community/logos.md?plain=1#L4).

Regardless, I think we should keep a section like this one near the top with high level feature highlights. On top of adding new things, I think it would be good to focus more on the value that this delivers for users (as opposed to leading with "RunInference takes advantage of existing Apache Beam concepts"). So maybe something like:

```
RunInference allows you to efficiently use models in your pipelines optimized for machine learning inferences. It includes the following out of the box features:

- Dynamic batching of inputs based on pipeline throughput to efficiently feed your model.
- A central model manager that determines the optimal number of models to load in order to balance memory and throughput usage
- Automatic mode refresh to ensure that your pipeline always uses the most recently deployed version of your model
- Support for many frameworks and model hubs, including Tensorflow, Pytorch, Sklearn, XGBoost, Hugging Face, TensorFlow Hub, Vertex AI, TensorRT, and ONNX
- Support for arbitrary frameworks using a custom model handler
- Support for multi-model pipelines
```

with links to the sections where we talk about each of these.

Comment (rszper) [reply website/www/site/content/en/documentation/sdks/python-machine-learning.md:59]: I agree with all of this. Most of the content already exists in both pages, but there are a few things that would need to move over. That's not a big deal, and I can make a PR for that if that's the direction we decide to go.

File: website/www/site/content/en/documentation/sdks/python-machine-learning.md [hunk 4]
```diff
@@ -47,9 +47,16 @@ For more infomation about machine learning with Apache Beam, see:
 * [About Beam ML](/documentation/ml/about-ml)
 * [RunInference notebooks](https://github.com/apache/beam/tree/master/examples/notebooks/beam-ml)
 
+## Support and limitations
+
+- The RunInference API is supported in Apache Beam 2.40.0 and later versions.
+- Model handlers are available for PyTorch, scikit-learn, TensorFlow, ONNX, and TensorRT. You can also use a custom model handler.
```
Comment (AnandInguva) [inline website/www/site/content/en/documentation/sdks/python-machine-learning.md:53]: Hugging Face, XGBoost, VertexAI
